### PR TITLE
New lookup module: manifold

### PR DESCRIPTION
**Pull request by galanoff**
##### SUMMARY

Add Manifold.co integration. The plugin fetches resource credentials from Manifold service.

##### ISSUE TYPE

- New Module Pull Request

### CODE REVIEW

**Comment by gundalow** [comment]
@galanoff Thank you for this PR, and fixing the issues that CI found.
Do you know anyone that would be able to do a functional review of this, perhaps someone at manifold.co?

**Comment by garyposter** [comment]
Thanks @gundalow and @galanoff . 

We've done a functional review prior to merging. @davidharrigan has volunteered to take a look at it again this week to doublecheck.

File: lib/ansible/plugins/lookup/manifold.py [hunk 1]
```diff
@@ -0,0 +1,276 @@
...
+        headers = default_headers
+        arg_headers = kwargs.pop('headers', None)
+        if arg_headers:
+            headers.update(arg_headers)
+
+        try:
+            display.vvvv('manifold lookup connecting to {0}'.format(url))
+            response = open_url(url, headers=headers, http_agent=self.http_agent, *args, **kwargs)
+            data = response.read()
+            if response.headers.getheader('content-type') == 'application/json':
```
Comment (davidharrigan) [inline lib/ansible/plugins/lookup/manifold.py]: This line fails for me trying to run Ansible under Python3 - `response.headers` is there, but there is no `getheader` method.  I don't know the internals of what is returned from `open_url` but I can call `get` on it and it works for both Python2 and Python3:

```
fatal: [127.0.0.1]: FAILED! => {
    "msg": "An unhandled exception occurred while running the lookup plugin 'manifold'. Error was a <class 'ansible.errors.AnsibleError'>, original message: ['Traceback (most recent call last):\\n', '  File \"/Users/dave/work/src/ansible-test/ansible/lib/ansible/plugins/lookup/manifold.py\", line 226, in run\\n    team_data = client.get_teams(team)\\n', '  File \"/Users/dave/work/src/ansible-test/ansible/lib/ansible/plugins/lookup/manifold.py\", line 166, in get_teams\\n    data = self.request(api, endpoint)\\n', '  File \"/Users/dave/work/src/ansible-test/ansible/lib/ansible/plugins/lookup/manifold.py\", line 113, in request\\n    if response.headers.getheader(\\'content-type\\') == \\'application/json\\':\\n', \"AttributeError: 'HTTPMessage' object has no attribute 'getheader'\\n\"]"
}
```

**Review by davidharrigan — COMMENTED** [review]
awesome work @galanoff ! I've verified this is functional under python2, but having trouble running this using python3.

File: lib/ansible/plugins/lookup/manifold.py (hunk 1 above)
Comment (gundalow) [reply lib/ansible/plugins/lookup/manifold.py]: @davidharrigan Nice spot, thanks.
From a quick Google I've found
* https://github.com/TAXIIProject/libtaxii/commit/59e18912e90550e2248779518fb63fac77d2f5a1
* https://bugs.python.org/issue4773

Comment (garyposter) [reply lib/ansible/plugins/lookup/manifold.py]: Thank you both! @galanoff plans to get those fixes in. @gundalow do we have a deadline for 2.8?

**Comment by galanoff** [comment]
Hi @gundalow , 

I have added a commit with a fix, however, it seems unrelated tests have failed.

**Comment by davidharrigan** [comment]
fwiw I have this working under python2 and python3 so I think we're good to go once those tests pass?

**Comment by gundalow** [comment]
rebuild_merge

Thanks all!

**Comment by galanoff** [comment]
ready_for_review

**Pull request merged**